File: Assets/2DWater/Scripts/C#/WaveSimulator.cs
using UnityEngine;
using System.Collections;

public class WaveSimulator : MonoBehaviour {

public float WaveSpeed;
public float WaveHeight;
public float OriginalY;

void  Start (){
	OriginalY = transform.position.y;
}

void  FixedUpdate (){
	transform.position = new Vector3(transform.position.x,OriginalY + Mathf.Sin(Time.time * WaveSpeed) * WaveHeight,transform.position.z);
	//if(transform.position.y < 0.5f){
	//	transform.position.y = 0.5f;
	//}
}
}

File: Assets/2DWater/Scripts/JS/WaveSimulator.js
#pragma strict
var WaveSpeed: float;
var WaveHeight: float;
var OriginalY: float;
function Start () {
	OriginalY = transform.position.y;
}

function FixedUpdate () {
	transform.position.y = OriginalY + Mathf.Sin(Time.time * WaveSpeed) * WaveHeight;
	//if(transform.position.y < 0.5){
	//	transform.position.y = 0.5;
	//}
}